### PR TITLE
Onboarding

### PR DESCRIPTION
**Pull request by rase-rocks**
Onboarding UI. Issue #13 

Things to note:

I have moved all strings into a Strings struct, ready for translation, is this something we can do at an app level or should we keep it on a per component basis?

### CODE REVIEW

File: CoEpi/ui/HealthQuiz/HealthQuizViewController.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+//  HealthQuizViewController.swift
+//  CoEpi
+//
+//  Created by Dusko Ojdanic on 3/26/20.
+//  Copyright © 2020 org.coepi. All rights reserved.
+//
+
+import UIKit
+
+class HealthQuizViewController: UIViewController {
```
Comment (johnsonh) [inline CoEpi/ui/HealthQuiz/HealthQuizViewController.swift]: Would it be possible to not touch/move the health quiz code? it'll create merge conflicts for me in https://github.com/Co-Epi/app-ios/pull/12

File: CoEpi/ui/onboarding/AttributedString.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        
+    }
+    
+    func append(_ value: String, attributes: FontAttributes) {
+        let attributedString    = NSAttributedString(string: "\(value) ", attributes: attributes)
+        storage.append(attributedString)
+    }
+    
+    func bold(_ value: String) -> AttributableString {
+
```
Comment (johnsonh) [inline CoEpi/ui/onboarding/AttributedString.swift]: Several extra newlines in these functions

File: CoEpi/ui/onboarding/Constants.swift [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+//  Copyright © 2020 org.coepi. All rights reserved.
+//
+
+import UIKit
+
+struct Constants {
+    
+    struct color {
+        
+        static let coepiBlue            = UIColor(named: "CoEpiBlue")!
```
Comment (johnsonh) [inline CoEpi/ui/onboarding/Constants.swift]: I would've suggested we stick with built in colors for now for simplicity, but given that this work is done, we might as well.

File: CoEpi.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -41,6 +41,31 @@
 		84FA01002427ED5E00318104 /* HomeViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 84FA00FE2427ED5E00318104 /* HomeViewController.swift */; };
 		84FA01012427ED5E00318104 /* HomeViewController.xib in Resources */ = {isa = PBXBuildFile; fileRef = 84FA00FF2427ED5E00318104 /* HomeViewController.xib */; };
 		CC71F7CAF64914F41BBE05C5 /* Pods_CoEpi_CoEpiUITests.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 4AB30FDD8E48688E1BF91F91 /* Pods_CoEpi_CoEpiUITests.framework */; };
+		CE98BE88242CE27D00EFB67A /* HealthQuizViewModel.swift in Sources */ = {isa = PBXBuildFile; fileRef = CE98BE87242CE27D00EFB67A /* HealthQuizViewModel.swift */; };
+		CE98BE8A242CE4E400EFB67A /* QuestionCell.swift in Sources */ = {isa = PBXBuildFile; fileRef = CE98BE89242CE4E400EFB67A /* QuestionCell.swift */; };
+		D48ADDC7242D315100E595D9 /* GradientView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D48ADDC6242D315100E595D9 /* GradientView.swift */; };
+		D48ADDC8242D315100E595D9 /* GradientView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D48ADDC6242D315100E595D9 /* GradientView.swift */; };
```
Comment (johnsonh) [inline CoEpi.xcodeproj/project.pbxproj]: Did GradientView get added multiple times?

Comment (johnsonh) [reply CoEpi.xcodeproj/project.pbxproj]: Looks like so for several o these? Strings, Constants, Assets, OnboardingButton, etc

File: CoEpi/ui/onboarding/OnboardingViewController.swift [hunk 5]
```diff
@@ -2,18 +2,110 @@ import UIKit
...
+    
+    @IBOutlet weak var gradientView     : GradientView!
+    @IBOutlet weak var getStartedButton : OnboardingButton!
+    @IBOutlet weak var privacyButton    : UIButton!
+    
+    private let viewModel               : OnboardingViewModel
 
     init(viewModel: OnboardingViewModel) {
-        self.viewModel = viewModel
+        self.viewModel      = viewModel
```
Comment (johnsonh) [inline CoEpi/ui/onboarding/OnboardingViewController.swift]: extra tab

File: CoEpi/ui/onboarding/Strings.swift [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+//
+//  Created by Robert Atherton on 26/03/2020.
+//  Copyright © 2020 org.coepi. All rights reserved.
+//
+
+import Foundation
+
+struct Strings {
+    
+    struct onboarding {
```
Comment (johnsonh) [inline CoEpi/ui/onboarding/Strings.swift]: `enum Onboarding`?

File: CoEpi/ui/onboarding/Strings.swift [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+//  Strings.swift
+//  CoEpi
+//
+//  Created by Robert Atherton on 26/03/2020.
+//  Copyright © 2020 org.coepi. All rights reserved.
+//
+
+import Foundation
+
+struct Strings {
```
Comment (johnsonh) [inline CoEpi/ui/onboarding/Strings.swift]: `enum Strings`? Since this is just being used as a namespace

**Comment by rase-rocks** [comment]
I have moved these changes to the project proper, so this fork is no longer needed

**Pull request closed without merging**